grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/services/impl/DefaultCleanJobRepository.groovy

```diff
@@ -0,0 +1,95 @@
+package com.twcable.grabbit.spring.batch.repository.services.impl
+
+import com.twcable.grabbit.jcr.JcrUtil
+import com.twcable.grabbit.spring.batch.repository.JcrJobRepositoryFactoryBean
+import com.twcable.grabbit.spring.batch.repository.services.CleanJobRepository
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.apache.felix.scr.annotations.Activate
+import org.apache.felix.scr.annotations.Component
+import org.apache.felix.scr.annotations.Reference
+import org.apache.felix.scr.annotations.Service
+import org.apache.sling.api.resource.PersistenceException
+import org.apache.sling.api.resource.Resource
+import org.apache.sling.api.resource.ResourceResolver
+import org.apache.sling.api.resource.ResourceResolverFactory
+import org.springframework.batch.core.BatchStatus
+import org.springframework.context.ConfigurableApplicationContext
+
+@Slf4j
+@CompileStatic
+@Component(label = "Grabbit Clean Job Repository Service", description = "Grabbit Clean Job Repository Service", immediate = true, metatype = true, enabled = true)
+@Service(CleanJobRepository)
+@SuppressWarnings(['GroovyUnusedDeclaration', 'GrMethodMayBeStatic'])
+class DefaultCleanJobRepository implements CleanJobRepository {
+
+    @Reference
+    ResourceResolverFactory resourceResolverFactory
+
+    @Reference
+    ConfigurableApplicationContext configurableApplicationContext
+
+    @Activate
+    void activate() {
+        log.info "CleanJobRepository Service activated"
+    }
+
+    @Override
+    void cleanJobRepository(int hours) {
+        JcrJobRepositoryFactoryBean jobRepositoryFactoryBean = configurableApplicationContext.getBean(JcrJobRepositoryFactoryBean)
+
+        if(!jobRepositoryFactoryBean) {
+            log.error "Cannot get an instance of JcrJobRepositoryFactoryBean. Will not clean up Grabbit Jcr Job Repository"
+            return
+        }
```

Comment:
Also added a short-circuit here. If this is null, nothing below is going to work :-)